src/clj_gatling/progress_tracker.clj

```diff
@@ -4,11 +4,16 @@
             (java.util.concurrent Executors TimeUnit)))
 
 (defn create-console-progress-tracker []
-  (let [finished? (atom false)]
+  (let [finished? (atom false)
+        width     (atom 0)]
     (fn [{:keys [progress sent-requests total-concurrency]}]
       (when-not @finished?
-        (let [progress-percent (int (* 100 progress))]
-          (print (str "Progress: " progress-percent "%, concurrency: " total-concurrency " , sent requests: " sent-requests "\r"))
+        (let [progress-percent (int (* 100 progress))
+              report-str (str "Progress: " progress-percent "%, concurrency: " total-concurrency " , sent requests: " sent-requests)
+              [previous-width new-width] (swap-vals! width (fn [_] (count report-str)))
+              length-diff (- previous-width new-width)
+              blanking (if (< 0 length-diff) (apply str (repeat length-diff " ")) "")]
+          (print report-str blanking "\r")
```

Comment:
This makes sure that this displays properly:
```
Progress: 18%, concurrency: 12, sent requests: 2031
Progress: 18%, concurrency: 8, sent requests: 2034
```

Before, the second line would end up displaying as
```
Progress: 18%, concurrency: 8, sent requests: 20341
```

Reply:
Good catch. This was a definitely a bug.